sdk/resourcemanager/internal/testutil/helper.go

```diff
@@ -53,3 +54,47 @@ func DeleteResourceGroup(ctx context.Context, subscriptionId string, cred azcore
 	rgClient := armresources.NewResourceGroupsClient(subscriptionId, cred, options)
 	return rgClient.BeginDelete(ctx, resourceGroupName, nil)
 }
+
+// CreateDeployment will create a resource using arm template.
+// It will return the deployment result entity.
+func CreateDeployment(ctx context.Context, subscriptionId string, cred azcore.TokenCredential, options *arm.ClientOptions, resourceGroupName, deploymentName string, template, params map[string]interface{}) (*armresources.DeploymentExtended, error) {
```

Comment:
This function has too many parameters - maybe we could introduce a wrapper for them? Or just use `armresource.Deployment` as a parameter to simplify this?
Is "avoiding explicitly import `armresources` package" one of the purpose of introducing this function?

Reply:
It is for scenario generation. There are two types of the step for API scenario. One is rest call, another is arm template.